test/test_other.py

```diff
@@ -10780,8 +10780,8 @@ def test(infile, source_map_added_dir=''):
 
     test('A ä☃ö Z.cpp')
 
-    ensure_dir('inner Z ö☃ä A')
-    test('inner Z ö☃ä A/A ä☃ö Z.cpp', 'inner Z ö☃ä A')
+    ensure_dir('inner Z ö☃ä A/21')
```

Comment:
Maybe add comment here as to why we need strange paths in the test?

Reply:
Good idea